tests/src/test_cases/test_mem.c

```diff
@@ -0,0 +1,40 @@
+#if LV_BUILD_TEST
+#include "../lvgl.h"
+
+#include "unity/unity.h"
+
+void setUp(void)
+{
+    /* Function run before every test */
+}
+
+void tearDown(void)
+{
+    /* Function run after every test */
+}
+
+/* Non-zero size with null pointer should behave like malloc */
+void test_realloc_safe_behave_malloc_like(void)
+{
+    void * ptr = NULL;
+    void * data = &ptr;
+    size_t allocation_size = sizeof(lv_font_t);
+
+    lv_res_t retval = lv_mem_realloc_safe(&data, allocation_size);
+    TEST_ASSERT_EQUAL_UINT16(retval, LV_RES_OK);
+
+    /* Don't leak memory */
+    lv_mem_free(ptr);
+}
+
+/* a zero size with a non-null pointer will behave like free and return LV_RES_OK. */
+void test_realloc_safe_behave_free_like(void)
+{
+    void * font = lv_mem_alloc(1 * sizeof(lv_font_t));
+    void * data = &font;
```

Comment:
Same here. proper test code should be

```c
void* font = lv_mem_alloc(1 * sizeof(lv_font_t));

//And then to realloc (or free in this case since len is 0)
lv_res_t retval = lv_mem_realloc_safe(&font, 0);

// font should now point to NULL as safe function should set it to NULL as it was free operation
if (font != NULL) {
    //Error
}
```

Reply:
I've refactored the test a bit, can you check 503e757?

Reply:
It currently sets the pointer to `zero_mem`, should it set it to NULL instead?

Reply:
How does 60b0f318 look? I think we could also move `ZERO_MEM_SENTINEL` to `lv_mem.h` to be able to use it on the tests, any thoughts @kisvegabor ?